sdk/metric/meter.go

```diff
@@ -77,7 +78,23 @@ func (m *meter) AsyncFloat64() asyncfloat64.InstrumentProvider {
 // RegisterCallback registers the function f to be called when any of the
 // insts Collect method is called.
 func (m *meter) RegisterCallback(insts []instrument.Asynchronous, f func(context.Context)) error {
-	m.pipes.registerCallback(f)
+	for _, inst := range insts {
+		switch t := inst.(type) {
```

Comment:
Does this switch work when used with the pre-registered instruments, e.g. [metric/internal.afCounter](https://github.com/open-telemetry/opentelemetry-go/blob/cc1eaec68316f938d6f008c76f96128c2f008b5d/metric/internal/global/instruments.go#L31)?  I believe the delegated MeterProvider will unwrap these, but it won't be able to if the MeterProvider is delegated between creating the instrument (resulting in the wrapped `afCounter`) and the RegisterCallback (resulting in this being called directly).

Reply:
I'll plan to update to identify the global instruments and I'll drop the return of an error.  Instead I'll default to "fail open" by allowing the registration to happen if we don't recognize the instrument type.